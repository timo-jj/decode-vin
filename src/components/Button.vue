<template>
  <button
    type="button"
    :class="[
      'button',
      {
        'button--disabled': disabled,
      },
    ]"
    :disabled="disabled"
    @click="onClick"
  >
    Lookup Vehicle
  </button>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue'

export default defineComponent({
  name: 'Button',

  props: {
    disabled: {
      type: Boolean as PropType<boolean>,
      deafult: false,
    },
  },

  emits: ['click'],

  methods: {
    onClick(): void {
      this.$emit('click')
    },
  },
})
</script>

<!-- script setup syntax
<script setup lang="ts">
import { defineEmits, defineProps, PropType } from 'vue'

defineProps({
  disabled: {
    type: Boolean as PropType<boolean>,
    deafult: false,
  },
})

const emit = defineEmits<{
  (eventName: 'click'): void
}>()

const onClick = (): void => {
  emit('click')
}
</script>
-->

<style lang="scss" scoped>
.button {
  border: 0;
  outline: 0;
  padding: 8px 24px;
  background: #1876d2;
  border-radius: 4px;
  color: #ffffff;
  text-transform: uppercase;
  box-shadow: rgba(9, 30, 66, 0.25) 0px 4px 8px -2px, rgba(9, 30, 66, 0.08) 0px 0px 0px 1px;
  font-weight: 700;
  cursor: pointer;
  transition: 0.2s ease-in-out;

  &--disabled {
    background: lightgray;
  }
}
</style>
