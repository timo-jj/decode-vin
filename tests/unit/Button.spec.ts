import { shallowMount } from '@vue/test-utils'

import Button from '@/components/Button.vue'

const wrapper = shallowMount(Button, {
  emits: ['click'],
  propsData: {
    disabled: false,
  },
} as any)

describe('Button', () => {
  describe('when rendering', () => {
    it('it renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('on click', () => {
    it('should emit click event', async () => {
      await wrapper.trigger('click')

      expect(wrapper.emitted('click')).toBeTruthy()
    })
  })

  describe('when button disabled', () => {
    beforeEach(() => {
      wrapper.setProps({
        disabled: true,
      })
    })

    it('disbaled class should be active', async () => {
      expect(wrapper.classes()).toContain('button--disabled')
    })
  })
})
