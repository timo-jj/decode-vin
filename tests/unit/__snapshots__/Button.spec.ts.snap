// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when rendering it renders correctly 1`] = `
<button
  class="button"
  type="button"
>
   Lookup Vehicle 
</button>
`;
